client/src/components/Form/Elements/FormCheck.vue

```diff
@@ -0,0 +1,48 @@
+<script setup>
+import { computed, ref } from "vue";
+
+const $emit = defineEmits(["input"]);
+const props = defineProps({
+    value: {
+        default: null,
+    },
+    options: {
+        type: Array,
+        required: true,
+    },
+});
+const selectAll = ref(false);
+
+const currentValue = computed({
+    get: () => {
+        const val = props.value || [];
```

Comment:
```suggestion
        const val = props.value ?? [];
```

`||` can behave unexpectedly when the left hand value is and empty string `""`, `0`, or `false`.
`??` checks for nullish values (`null` or `undefined`) only, avoiding edge cases.

See [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#assigning_a_default_value_to_a_variable) for more information

Reply:
Historically we treated `""` as `null` but that was a mistake and caused issues down the road. Moving forward we should definitely follow the advice here to improve the code reliability. Thank you.